src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -549,4 +554,14 @@ static <T> AbstractRowBuffer<T> createRowBuffer(
             ErrorCode.INTERNAL_ERROR, "Unsupported BDEC format version: " + bdecVersion);
     }
   }
+
+  private static void batchSizeWarning(long rowsPassed) {
+    if (rowsPassed > RECOMMENDED_MAX_BATCH_SIZE) {
```

Comment:
I'm not sure if we want to warn on row count, should we use batch data size instead? Also this is case by case, a large batch would work fine if you have enough resources so I'm not sure we should warn on anything, it just adds noise. Updating our online doc should be sufficient?

Reply:
+1, and can you add more about why we chose 10k rows?

Reply:
You are right, row count does not matter, only batch size does. We discussed it with @sfc-gh-azagrebin offline and this PR now does the following If the batch size passed to insertRows() is:

1. more than 128 MB => throw an exception
2. more than 16 MB => emit warning log

Both values are configurable by the user.